flexprep/domain/db_utils.py

```diff
@@ -0,0 +1,112 @@
+import logging
+import sqlite3
+from datetime import datetime as dt
+
+from flexprep import CONFIG
+from flexprep.domain.data_model import IFSForecast
+
+
+class DB:
+    conn: sqlite3.Connection
+
+    def __init__(self) -> None:
+        try:
+            """Establish a database connection."""
+            self.db_path = CONFIG.main.db_path
+            self.conn = sqlite3.connect(self.db_path)
+            logging.info("Connected to database.")
+            self._initialize_db()
+        except sqlite3.Error as e:
+            logging.error(f"An error occurred: {e}")
```

Comment:
I assume that the intention is just to log the exception, a raise is needed to let it continue to bubble up
```suggestion
            logging.error(f"An error occurred: {e}")
            raise
```

Reply:
I've now added a raise in every except block